Description: iptables intrapositioned negation fixe for MAC
 Fix iptables intrapositioned negation --mac-source.
 Thanks to Alexander Prokopyev for reporting the issue
 and suggesting a patch; thanks to Phil Whineray
 <phil@sanewall.org> for confirming the patch
 and backporting the Sanewall patch to FireHOL.
Origin: debian
Author: Phil Whineray <phil@sanewall.org>
Last-Update: 2013-05-05

--- a/firehol.sh
+++ b/firehol.sh
@@ -2412,8 +2412,8 @@
 	
 	set_work_function "If the source IP ${1} does not match MAC ${2}, drop the packet"
 	
-	iptables -t filter -A INPUT   -s "${1}" -m mac --mac-source ! "${2}" -j WRONGMAC
-	iptables -t filter -A FORWARD -s "${1}" -m mac --mac-source ! "${2}" -j WRONGMAC
+	iptables -t filter -A INPUT   -s "${1}" -m mac ! --mac-source "${2}" -j WRONGMAC
+	iptables -t filter -A FORWARD -s "${1}" -m mac ! --mac-source "${2}" -j WRONGMAC
 	
 	return 0
 }
